pvlib/way_faster.py

```diff
@@ -0,0 +1,240 @@
+"""
+Faster ways to calculate single diode model currents and voltages using
+methods from J.W. Bishop (Solar Cells, 1988).
+"""
+
+from collections import OrderedDict
+import numpy as np
+try:
+    from scipy.optimize import brentq, newton
+except ImportError:
+    brentq = NotImplemented
+    newton = NotImplemented
+
+# TODO: update pvsystem.i_from_v and v_from_i to use "gold" method by default
+
+
+def est_voc(photocurrent, saturation_current, nNsVth):
+    """
+    Rough estimate of open circuit voltage useful for bounding searches for
+    ``i`` of ``v`` when using :func:`~pvlib.pvsystem.singlediode`.
+
+    :param numeric photocurrent: photo-generated current [A]
+    :param numeric saturation_current: diode one reverse saturation current [A]
+    :param numeric nNsVth: product of thermal voltage ``Vth`` [V], diode
+        ideality factor ``n``, and number of series cells ``Ns``
+    :returns: rough estimate of open circuit voltage [V]
+
+    The equation is from [1].
```

Comment:
Probably good to mention here that this is the Voc calculation for an ideal device with zero series resistance and infinite shunt resistance.

Reply:
✔️